src/ServiceControl/ExternalIntegrations/FallbackToOldContractTypes.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace ServiceControl.ExternalIntegrations
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+
+    public class FallbackToOldContractTypes : Behavior<IOutgoingPhysicalMessageContext>
+    {
+        public override Task Invoke(IOutgoingPhysicalMessageContext context, Func<Task> next)
+        {
+
+            if (context.Headers.TryGetValue(EnclosedTypesHeaderName, out var qualifiedTypeName))
+            {
+                if (qualifiedTypeName.EndsWith(NewAssemblyIdentifier))
+                {
+                    var typeName = qualifiedTypeName.Split(',')[0];
+
+                    context.Headers[EnclosedTypesHeaderName] = $"{typeName}, {OldAssemblyIdentifier}";
```

Comment:
How about we always just remove the version number?

Reply:
I wanted to make sure that the state is exactly as it was before we switched to ServiceControlContracts 3.0.

Reply:
I think by removing the version number we would make the contracts less restricted so that would be a non-breaking change for the existing customers.

Reply:
@tmasternak The version number can be removed, I've done that plenty of times (i.e. https://gist.github.com/ramonsmits/0ead15453e3bc1843cdc7f97718b5244)

I've never seen that break deserialization.